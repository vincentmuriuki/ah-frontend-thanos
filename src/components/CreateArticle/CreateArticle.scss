@import url('https://fonts.googleapis.com/css?family=Roboto+Condensed');

#titleInput, 
#descriptionInput,
#image_urlInput,
#tag_listInput {
    height: 40px;
    border-top: none;
    border-left: none; 
    border-right: none;
}

#createArticleBt {
    padding: 15px;
    width: 30%;
    background-color: #47d79f;
    border: none;
    font-family: "Roboto Condensed", sans-serif;
    &:hover {
        background-color: #3db184;
      }
    
}
.artcileCreate{
    font-family: "Roboto Condensed", sans-serif;
}
.createArticleTitle {
    font-family: "Roboto Condensed", sans-serif;
}

.artcileCreate {
    padding-bottom: 30px;
}
#bodyInput {
    height: 500px;
}
.create-article-form {
  width: 80%;
}

.image-upload{
    color: turquoise !important;
    border-color: turquoise !important; }

.image {
    color: white;
}
input[type="file"] {
    border-color: turquoise !important; 
}
