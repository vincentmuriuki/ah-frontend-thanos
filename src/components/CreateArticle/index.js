/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import './CreateArticle.scss';
import * as PropTypes from 'prop-types';


const formInputs = (inputName, onChange, inputId, minLength, maxLength) => (
  <div className="form-group">
    <textarea className="form-control" minLength={minLength} maxLength={maxLength} name={inputName} onChange={onChange} id={inputId} rows="3" required />
  </div>
);

const Image = onClick => (
  <div className="col-lg-8 mt-2">
    <input type="file" className="image" name="image_url" onClick={onClick} />
  </div>
);

const CreateArticle = ({ onChange, onSubmit, onClick }) => (
  <div>
    <br />
    <h3 className="text-center createArticleTitle">Create Your Article</h3>
    <hr className="col-sm-8 " />
    <br />
    <div className="col-sm-10 offset-2 artcileCreate">
      <form className="create-article-form" onSubmit={onSubmit}>
        <h5>Title: </h5>
        <small>Title must be 130 characters maximum and 5 characters minimum</small>
        {formInputs('title', onChange, 'titleInput', '5', '130')}
        <h5>Description</h5>
        <small>Title must be 150 characters maximum and 100 characters minimum</small>
        {formInputs('description', onChange, 'descriptionInput', '100', '120')}

        <h5>Tag list</h5>
        {formInputs('tag_list', onChange, 'tag_listInput')}
        <br />
        <h5>Upload Image</h5>
        {Image(onClick)}
        <br />
        <h5>Body</h5>
        {formInputs('body', onChange, 'bodyInput')}
        <div className="signin-createArticleBt">
          <button
            type="submit"
            className="btn btn-primary"
            id="createArticleBt"
            name="createArticleBt"
          >
              Publish Article
          </button>
        </div>
      </form>
    </div>
  </div>
);

CreateArticle.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default CreateArticle;
