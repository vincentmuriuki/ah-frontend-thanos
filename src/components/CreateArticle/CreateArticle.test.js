import React from 'react';
import { shallow } from 'enzyme';
import CreateArticle from './index';

describe('<CreateArticle /> ', () => {
  const CreateArticleComponent = shallow(
    <CreateArticle
      onChange={jest.fn()}
      onSubmit={jest.fn()}
      onClick={jest.fn()}
    />,
  );
  test('renders the component', () => {
    expect(CreateArticleComponent).toMatchSnapshot();
  });
});
