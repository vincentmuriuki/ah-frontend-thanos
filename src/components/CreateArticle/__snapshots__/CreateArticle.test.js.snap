// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateArticle />  renders the component 1`] = `
<div>
  <br />
  <h3
    className="text-center createArticleTitle"
  >
    Create Your Article
  </h3>
  <hr
    className="col-sm-8 "
  />
  <br />
  <div
    className="col-sm-10 offset-2 artcileCreate"
  >
    <form
      className="create-article-form"
      onSubmit={[MockFunction]}
    >
      <h5>
        Title: 
      </h5>
      <small>
        Title must be 130 characters maximum and 5 characters minimum
      </small>
      <div
        className="form-group"
      >
        <textarea
          className="form-control"
          id="titleInput"
          maxLength="130"
          minLength="5"
          name="title"
          onChange={[MockFunction]}
          required={true}
          rows="3"
        />
      </div>
      <h5>
        Description
      </h5>
      <small>
        Title must be 150 characters maximum and 100 characters minimum
      </small>
      <div
        className="form-group"
      >
        <textarea
          className="form-control"
          id="descriptionInput"
          maxLength="120"
          minLength="100"
          name="description"
          onChange={[MockFunction]}
          required={true}
          rows="3"
        />
      </div>
      <h5>
        Tag list
      </h5>
      <div
        className="form-group"
      >
        <textarea
          className="form-control"
          id="tag_listInput"
          name="tag_list"
          onChange={[MockFunction]}
          required={true}
          rows="3"
        />
      </div>
      <br />
      <h5>
        Upload Image
      </h5>
      <div
        className="col-lg-8 mt-2"
      >
        <input
          className="image"
          name="image_url"
          onClick={[MockFunction]}
          type="file"
        />
      </div>
      <br />
      <h5>
        Body
      </h5>
      <div
        className="form-group"
      >
        <textarea
          className="form-control"
          id="bodyInput"
          name="body"
          onChange={[MockFunction]}
          required={true}
          rows="3"
        />
      </div>
      <div
        className="signin-createArticleBt"
      >
        <button
          className="btn btn-primary"
          id="createArticleBt"
          name="createArticleBt"
          type="submit"
        >
          Publish Article
        </button>
      </div>
    </form>
  </div>
</div>
`;
